src/NUnitEngine/Portable/nunit.portable.agent/ResultSummary.cs

```diff
@@ -0,0 +1,342 @@
+
+        /// <summary>
+        /// Gets the number of asserts
+        /// </summary>
+        public int AssertCount { get; private set; }
+
+        /// <summary>
+        /// The overall result of the test run
+        /// </summary>
+        public string Result { get; private set; }
+
+        /// <summary>
+        /// The start of the test run
+        /// </summary>
+        public DateTime StartTime { get; private set; }
+
+        /// <summary>
+        /// The end of the test run
+        /// </summary>
+        public DateTime EndTime { get; private set; }
+
+        /// <summary>
+        /// The length of the test run in seconds
+        /// </summary>
+        public double Duration { get; private set; }
+
+        #endregion
+
+        #region Helper Methods
+
+        void InitializeCounters()
+        {
+            TestCount = 0;
+            PassCount = 0;
+            FailureCount = 0;
+            ErrorCount = 0;
+            InconclusiveCount = 0;
+            SkipCount = 0;
+            IgnoreCount = 0;
+            ExplicitCount = 0;
+            InvalidCount = 0;
+            InvalidAssemblies = 0;
+            AssertCount = 0;
+        }
+
+        void Summarize(IEnumerable<XElement> elements)
+        {
+            foreach (XElement element in elements)
+                Summarize(element);
+        }
+
+        void Summarize(XElement element)
+        {
+            string type = element.Attribute("type")?.Value;
+            string status = element.Attribute("result")?.Value;
+            string label = element.Attribute("label")?.Value;
+            string result = element.Attribute("result")?.Value;
+
+            switch (element.Name.ToString())
+            {
+                case "test-case":
+                    TestCount++;
+
+                    switch (status)
+                    {
+                        case "Passed":
+                            PassCount++;
+                            break;
+                        case "Failed":
+                            if (label == null)
+                                FailureCount++;
+                            else if (label == "Invalid")
+                                InvalidCount++;
+                            else
+                                ErrorCount++;
+                            break;
+                        case "Inconclusive":
+                            InconclusiveCount++;
+                            break;
+                        case "Skipped":
+                            if (label == "Ignored")
+                                IgnoreCount++;
+                            else if (label == "Explicit")
+                                ExplicitCount++;
+                            else
+                                SkipCount++;
+                            break;
+                        default:
+                            SkipCount++;
+                            break;
+                    }
+                    break;
+
```

Comment:
Good catch. I also need to apply those changes to the `dotnet-test-nunit` project. That highlights a good reason why we need this class for those other projects like Xamarin and .NET Core, keep it all in one place. I wonder if we could somehow merge the three similar but slightly different `ResultSummary` classes?